### ISSUE TITLE
Placing restrictions on the agent for trading

### ISSUE THREAD

**Issue by codecolony** [opendilab/DI-engine#643]
- [ ] I have marked all applicable categories:
    + [ ] exception-raising bug
    + [ ] RL algorithm bug
    + [ ] system worker bug
    + [ ] system utils bug
    + [ ] code design/refactor
    + [ x] documentation request
    + [ ] new feature request
- [x ] I have visited the [readme](https://github.com/opendilab/DI-engine/blob/github-dev/README.md) and [doc](https://opendilab.github.io/DI-engine/)
- [ x] I have searched through the [issue tracker](https://github.com/opendilab/DI-engine/issues) and [pr tracker](https://github.com/opendilab/DI-engine/pulls)
- [ ] I have mentioned version numbers, operating system and environment, where applicable:
  ```python
  import ding, torch, sys
  print(ding.__version__, torch.__version__, sys.version, sys.platform)
  ```

Hi,

First of all, thank you very much for creating this incredible engine. I'm having a lot of fun tinkering with it.
Just trained my first model on Google Stocks. It did fine for what it was supposed to do given the env and player actions. 

I want to re-train with following conditions on the agent.

1. Maximum 'X' trades allowed per day. Here X could be 3 or 5, as example.
2. Any positions held should be squared off by end of day.

My trading data consists of either 1 minute or 5 minute candles spread across months.
I've checked trading_env.py for hints but I couldn't find much relevancy for my use case.

Any help in this regard would be very helpful.
Thanks in advance.

**Comment by Cloud-Pku** [comment]
Good advice. You're making this environment more realistic.

- For the first condition, we initially designed the action space as [double sell, sell, hold, buy, double buy], which has the advantage of reducing the position status, and for the profit, there is no need to record the actual amount, but the ratio to the initial capital. This is easy to align with the objective function of reinforcement learning.
If you want to change, I think you can redesign the action space as the following discrete form: [..., -2, -1, hold, 1, 2,... X]. But the premise is that the agent has sufficient funds to support X trades at present. Maybe you need a special state to save the actual amount of funds at present. In addition, 3 states are not enough to describe the position status, perhaps it needs to be designed as [..., -2, -1, 0, 1...]

- For the second condition, you need to record the current time with a time variable so that at the end of each day, the Flat operation is done inside env and the Flat state is passed to the agent on the following day.

In summary, I think you need to redesign the action space and the state space, clarify the state transition process, and align the cumulative profit and reinforcement learning optimization goals by designing appropriate reward functions.
